olive/evaluator/metric.py

```diff
@@ -39,78 +42,13 @@ class LatencySubType(str, Enum):
-# names, subtypes: Union[str, List[str]]
-# However accuracy metric poses a slight problem since AUC has a different config. Need to resolve this
-# so that we get a single metric config for a single type
-# This way, the user can return multiple metrics at once
-class Metric(ConfigBase):
-    name: str
-    type: MetricType
-    sub_type: Union[AccuracySubType, LatencySubType] = None
-    higher_is_better: bool = True
-    priority_rank: int = 1
-    goal: MetricGoal = None
+class MetricSubType(ConfigBase):
+    name: Union[AccuracySubType, LatencyMetricConfig, str]
     metric_config: ConfigBase = None
-    user_config: ConfigBase = None
-    data_config: DataConfig = DataConfig()
-
-    @validator("sub_type", always=True, pre=True)
-    def validate_sub_type(cls, v, values):
-        if "type" not in values:
-            raise ValueError("Invalid type")
-
-        if values["type"] == MetricType.CUSTOM:
-            return None
-        sub_type_enum = AccuracySubType if values["type"] == MetricType.ACCURACY else LatencySubType
-        try:
-            v = sub_type_enum(v)
-        except ValueError:
-            raise ValueError(
-                f"sub_type must be one of {list(sub_type_enum.__members__.keys())} for {values['type']} metric"
-            )
-        return v
-
-    @validator("higher_is_better", always=True, pre=True)
-    def validate_higher_is_better(cls, v, values):
-        if "type" not in values:
-            raise ValueError("Invalid type")
-
-        if values["type"] == MetricType.ACCURACY:
-            return True
-        if values["type"] == MetricType.LATENCY:
-            return False
-        if v is None:
-            raise ValueError("higher_is_better must be specified for custom metric")
-        return v
-
-    @validator("metric_config", always=True, pre=True)
-    def validate_metric_config(cls, v, values):
-        if "type" not in values:
-            raise ValueError("Invalid type")
-        if "sub_type" not in values:
-            raise ValueError("Invalid sub_type")
-
-        if values["type"] == MetricType.CUSTOM:
-            return None
-
-        # metric config class
-        if values["type"] == MetricType.LATENCY:
-            metric_config_class = LatencyMetricConfig
-        elif values["type"] == MetricType.ACCURACY:
-            metric_config_class = AccuracyBase.registry[values["sub_type"]].get_config_class()
-
-        # validate metric config
-        return validate_config(v, ConfigBase, metric_config_class)
-
-    @validator("user_config", pre=True)
-    def validate_user_config(cls, v, values):
-        if "type" not in values:
-            raise ValueError("Invalid type")
-
-        user_config_class = get_user_config_class(values["type"])
-        return validate_config(v, ConfigBase, user_config_class)
+    # -1 means no priority which will be evaluated only
+    priority_rank: int = -1
```

Comment:
if the default value is `-1`, I think we need to validate somewhere that atleast one type-subtype has value 1. Or else we won't have any signal for the search algorithms or metrics to rank output models by. 

Maybe we keep the default as before? assumed as 1 if only one type-subtype is present in the entire metric list, else, require user to provide unique rank or -1 for all type-subtypes

Reply:
Still keep the default value as -1 to avoid user to provide rank for those metrics they do not want to compare. The behavior will be like
1. When there is only sub metric, user can provide empty priority rank olive will assign as 1. (done in validator
2. When there is multi sub metrics like bert example, user can just assign the rank value to the metrics they care. For other metrics, -1 will be assign.